util/helpers.py

```diff
@@ -1,8 +1,147 @@
+        if any('Adam' not in v.op.name for v in init_vars):
+            log_error('Tried to load a CuDNN RNN checkpoint but there were '
+                      'more missing variables than just the Adam moment '
+                      'tensors.')
+            sys.exit(1)
+
+    if drop_source_layers>0:
+        '''
+        The transfer learning approach here needs us to supply
+        the layers which we will to exclude from the source model.
+        Say we want to exclude all layers except for the first one,
+        we are dropping five layers total, so:
+        drop_source_layers=5
+        If we want to use all layers from the source model except the last one, we use this:
+        drop_source_layers=1
+        '''
+        if drop_source_layers>=6:
+            log_error('The model only has 6 layers, but you are trying to drop '
+                      'all of them or more than all of them. Continuing and '
+                      'dropping only 5 layers')
+        
+        dropped_layers = ['2', '3', 'lstm', '5', '6'][-1 * min(5, int(drop_source_layers)):]
+
+        # Initialize all variables needed for DS, but not loaded from ckpt
+        for v in load_vars:
+            if any(layer in v.op.name for layer in dropped_layers):
+                init_vars.add(v)
+                
+        load_vars -= init_vars
+
+    for v in init_vars:
+        print("Initializing variable from scratch:", v.op.name)
+    for v in load_vars:
+        print("Loading variable from model:", v.op.name)
+        
+    init_op = tfv1.variables_initializer(list(init_vars))
+    for v in list(load_vars):
+        v.load(ckpt.get_tensor(v.op.name), session=session)
+    session.run(init_op)
+
+
+def check_model(checkpoint_filename):
+    checkpoint = tf.train.get_checkpoint_state(FLAGS.checkpoint_dir, checkpoint_filename)
+    if not checkpoint:
+        return False
+    return checkpoint.model_checkpoint_path
+
+
+def try_model(session, load_flag):
+
+    '''
+    if auto, cascasde from last --> best --> init
+  
+    if last, try last, if not found , break
+    
+    '''
+    def try_last():
+        return check_model('checkpoint')
+        
+    def try_best():
+        return check_model('best_dev_checkpoint')
+        
+    def try_init():
+        log_info('Initializing variables...')
+        # Initialize a new model from scratch
+        session.run(tfv1.global_variables_initializer())
+        return True
+    
+    def try_auto():
+        res = try_last()
+        if not res:
+            res = try_best()
+            if not res:
+                res = try_init()
+        return res
+    
+    checkpoint_path = False
+    if load_flag == 'last':
+        # returns false or a checkpoint_path
+        checkpoint_path = try_last()
+        if not checkpoint_path:
+            log_error('Unable to load LAST model from specified checkpoint dir'
+                      ' - consider using load option "auto" or "init".')
```

Comment:
I know you didn't write this message, but we should not be advising people to just blindly try different `--load` values. Specially as this can be a destructive action with `init`. And maybe include the path in the message: `'Unable to load LAST model from specified checkpoint directory "{}".'.format(FLAGS.checkpoint_dir)`

Reply:
I agree, and I've changed the error log to include the dir as you mention